### PR TITLE
Run workflow on finished prowjob events.

### PR DESCRIPTION
**Pull request by dekiel**
Using condition nested steps for better flow representation.

**Description**

Changes proposed in this pull request:

Added step and condition to check if event is for finished prowjob. Workflow will not call services for not finished prowjobs.
Refactored workflow code to better represent flow and decision steps.

### CODE REVIEW

File: development/gcp/workflows/secrets-leak-detector.yaml [hunk 1]
```diff
@@ -1,69 +1,79 @@
...
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-            X-Cloud-Trace-Context: $${map.get(step_result.headers, "X-Cloud-Trace-Context")}
-          body: $${step_result.body}
-        result: step_result
-    - return_result:
-        return: $${step_result}
+          - condition: $${step_result.status == "success" or step_result.status == "failure" or step_result.status == "aborted"}
```
Comment (akiioto) [inline development/gcp/workflows/secrets-leak-detector.yaml:11]: Is it possible to simplify this condition to sth like this? Would be easier to read and/or extend in future.

condition: $${step_result.status in ["success", "failure", "aborted"]}

Comment (dekiel) [reply development/gcp/workflows/secrets-leak-detector.yaml]: According to documentation it should work.

**Comment by dekiel** [comment]
Applied review suggestion.

**Review by akiioto — APPROVED** [review]

**Pull request merged**